conans/client/conf/detect.py

```diff
@@ -127,8 +129,11 @@ def _detect_compiler_version(result, output, profile_path):
         if compiler == "apple-clang":
             result.append(("compiler.libcxx", "libc++"))
         elif compiler == "gcc":
-            result.append(("compiler.libcxx", "libstdc++"))
-            if Version(version) >= Version("5.1"):
+            new_abi_available = Version(version) >= Version("5.1")
+            libcxx, old_abi = ('libstdc++11', False) if new_abi_available and get_env(CONAN_V2_MODE_ENVVAR, False)\
```

Comment:
I agree there is not a direct implication, but it is also not true the other way around

IMO this is the right way to go, this is what we have in the trainings, this is what many people are doing right after installing Conan, this is the origin of some issues and... this removes the big warning you see just after running Conan for the first time: `WARNING: GCC OLD ABI COMPATIBILITY` which scares a little bit.

Reply:
Ok, yes, I agree it is better, because it is much more likely than the opposite. Good, leave it as-is, but please check if there is a ticket for Conan 2.0 already, and otherwise create it, Conan shouldn't incorrectly set libstdc++11 if that is not the current one

Reply:
This one should be enough: https://github.com/conan-io/conan/issues/2340

> It will be almost sure one of the things that will be changed for future 2.0, but still some time for that.